packages/jest-runtime/src/transform.js

```diff
@@ -67,15 +63,15 @@ const getCacheKey = (
       testRegex: config.testRegex,
     }));
   }
-  const configStr = configToJsonMap.get(config) || '';
+  const confStr = configToJsonMap.get(config) || '';
   const preprocessor = getPreprocessor(config);
 
   if (preprocessor && typeof preprocessor.getCacheKey === 'function') {
-    return preprocessor.getCacheKey(fileData, filePath, configStr);
+    return preprocessor.getCacheKey(fileData, filePath, confStr, {instrument});
   } else {
     return crypto.createHash('md5')
       .update(fileData)
-      .update(configStr)
+      .update(confStr)
       .digest('hex');
```

Comment:
instrument needs to be added to the default cache key here too. This is used when the preprocessor doesn't expose its own cache key function.